GitCommands/Settings/Setting.cs

```diff
@@ -173,7 +173,24 @@ private object GetValue(string name)
                 return SettingsSource
                     .GetValue<object>(name, null, value =>
                     {
-                        switch (Type.GetTypeCode(typeof(T)))
+                        var type = typeof(T);
+                        var underlyingType = Nullable.GetUnderlyingType(type) ?? type;
+
+                        if (underlyingType.IsEnum)
+                        {
+                            try
+                            {
+                                var result = Enum.Parse(underlyingType, value);
```

Comment:
Why not https://docs.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=netframework-4.6.1?

Reply:
.NET Core 3.1

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=netframework-4.6.1

Reply:
where TEnum : struct;

![image](https://user-images.githubusercontent.com/3169265/94532365-72af7600-0246-11eb-88f1-6f196a7c9e02.png)